azurecompute-arm/src/test/java/org/jclouds/azurecompute.arm/features/ResourceGroupApiLiveTest.java

```diff
@@ -0,0 +1,134 @@
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus.JobStatus;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+
+
+@Test(groups = "live", testName = "ResourceGroupApiLiveTest")
+public class ResourceGroupApiLiveTest extends BaseAzureComputeApiLiveTest {
+   private String resourcegroup;
+
+   @BeforeClass
+   @Override
+   public void setup(){
+      super.setup();
+      resourcegroup = getResourceGroupName();
+   }
+
+   private ResourceGroupApi api() {
+      return api.getResourceGroupApi();
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testList() {
+      final List<ResourceGroup> resourceGroups = api().list();
+
+      assertTrue(resourceGroups.size() > 0);
+
+      assertTrue(Iterables.any(resourceGroups, new Predicate<ResourceGroup>() {
+
+         @Override
+         public boolean apply(final ResourceGroup group) {
+            return resourcegroup.equals(group.name());
+         }
+      }));
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testRead() {
+      final ResourceGroup group = api().get(resourcegroup);
+      assertNotNull(group);
+      assertEquals(group.name(), resourcegroup);
+      assertEquals(group.location(), LOCATION);
+   }
+
+   public void testCreate() {
+
+      final ResourceGroup resourceGroup = api().create("jcloudstest", LOCATION, null);
+      assertEquals(resourceGroup.name(), "jcloudstest");
+      assertEquals(resourceGroup.location(), LOCATION);
+      assertEquals(resourceGroup.tags().size(), 0);
+      assertTrue(resourceGroup.id().contains("jcloudstest"));
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testUpdateWithEmptyTag() {
+      ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder().build();
+
+      final ResourceGroup resourceGroup = api().update("jcloudstest", tags);
+
+      assertEquals(resourceGroup.tags().size(), 0);
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testUpdateWithTag() {
+      ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder().put("test1", "value1").build();
+
+      final ResourceGroup resourceGroup = api().update("jcloudstest", tags);
+
+      assertEquals(resourceGroup.tags().size(), 1);
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void testDelete() throws Exception {
+      URI uri =  api().delete(resourcegroup);
+      assertNotNull(uri);
```

Comment:
This is sensible... If the resource could be deleted without generating a task (200 response) this will actually be `null` but the delete operation succeeded. We should only do the polling if the URI is not null